### ISSUE TITLE
error with Cluster_Blast_Extract.py

### ISSUE THREAD

**Issue by ptitle** [dportik/SuperCRUNCH#3]
I get the following error when running Cluster_Blast_Extract.py. This happens with one of the fasta files provided in this github, so I know it's not specific to my own fasta files:

All that is in the directory `~/testing` is `Squamate_12S_References.fasta`.
```
[ptitle@iu-eri-006207 testing]$ python ~/SuperCRUNCH/supercrunch-scripts/Cluster_Blast_Extract.py -i ~/testing -b dc-megablast -m span


=====================================================

Performing cd-hit-est to cluster sequences

=====================================================


================================================================
Program: CD-HIT, V4.8.1, May 06 2019, 16:09:45
Command: cd-hit-est -i Squamate_12S_References.fasta -o
         Squamate_12S_References_Out -c 0.8 -n 4 -M 16000 -d 0

Started: Mon May  6 16:30:16 2019
================================================================
                            Output                              
----------------------------------------------------------------
total seq: 115
longest and shortest : 962 and 781
Total letters: 107033
Sequences have been sorted

Approximated minimal memory consumption:
Sequence        : 0M
Buffer          : 1 X 12M = 12M
Table           : 1 X 0M = 0M
Miscellaneous   : 0M
Total           : 12M

Table limit with the given memory limit:
Max number of representatives: 4000000
Max number of word counting entries: 1998452794

comparing sequences from          0  to        115

      115  finished         51  clusters

Approximated maximum memory consumption: 12M
writing new database
writing clustering information
program completed !

Total CPU time 0.54







======================================================

Parsing results and writing clustered fasta files

======================================================


Parsing clusters for Squamate_12S_References_Out.clstr.

Traceback (most recent call last):
  File "/home/ptitle@ads.iu.edu/SuperCRUNCH/supercrunch-scripts/Cluster_Blast_Extract.py", line 782, in <module>
    main()
  File "/home/ptitle@ads.iu.edu/SuperCRUNCH/supercrunch-scripts/Cluster_Blast_Extract.py", line 756, in main
    Parse_Clusters(cluster_dir, f, parse_dir, args.in_dir)
  File "/home/ptitle@ads.iu.edu/SuperCRUNCH/supercrunch-scripts/Cluster_Blast_Extract.py", line 298, in Parse_Clusters
    gene_dict = get_seq_dict(main_dir, cluster_dir, gene_name)
  File "/home/ptitle@ads.iu.edu/SuperCRUNCH/supercrunch-scripts/Cluster_Blast_Extract.py", line 256, in get_seq_dict
    return record_dict
UnboundLocalError: local variable 'record_dict' referenced before assignment
```
I should have all dependencies properly installed. I can successfully do `import Bio` in python, and I can call cd-hit-est, blastn and makeblastdb from the command line.  

Any ideas?

**Comment by dportik** [comment]
I was able to reproduce the error exactly, and it looks like the problem occurs because of how the input fasta file is labeled. The Cluster_Blast_Extract.py script unfortunately has to use the underscores present in the various intermediate file names to obtain information. I renamed the input file `Squamate_12S_References.fasta` to `Squamate-12S-References.fasta`, and the error did not occur. If you can confirm this also solves your issue, I will make the instructions more explicit regarding file name labeling. To work properly, there should not be any underscores or periods present in the name portion of [NAME].fasta for the input files.

**Comment by ptitle** [comment]
Yeah, that solves it! In this particular case, the script ran into an error again, but due to the same label being present more than once in the fasta file. But I tried it on another file where I knew that was not an issue, and the script ran successfully. Thanks!

**Comment by dportik** [comment]
Thanks for pointing out the issue with the reference file - looks like a different version made it to the repository than what I have here. I went ahead and removed the duplicate entry and replaced the underscores with dashes in the file name (and checked all the other reference sequence sets). All the squamate references should now be compatible with this step. I also updated the instructions in the wiki to be explicit about file naming for Cluster_Blast_Extract.py. Hopefully this avoids any trouble in future analyses!
